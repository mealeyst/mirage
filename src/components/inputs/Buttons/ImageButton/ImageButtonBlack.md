```js
<ImageButtonBlack>Click Me!</ImageButtonBlack>
```

## With Arrow
```js
<ImageButtonBlack arrow>Click Me!</ImageButtonBlack>
```
