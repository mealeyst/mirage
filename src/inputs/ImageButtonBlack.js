import PropTypes from 'prop-types'
import styled from 'styled-components'

import BaseImageButton from './ImageButton.base'

const ImageButtonBlack = styled(BaseImageButton)`
  > button {
    background-color: ${props => props.theme.colors.black};
    color: ${props => props.theme.colors.white};
  }
  > .borderBottom {
    background-color: ${props => props.theme.colors.yellow};
  }
`

ImageButtonBlack.propTypes = {
  theme: PropTypes.shape({
    colors: PropTypes.shape({
      black: PropTypes.string,
      yellow: PropTypes.string,
      white: PropTypes.string
    })
  }).isRequired
}

/** @component */
export default ImageButtonBlack
