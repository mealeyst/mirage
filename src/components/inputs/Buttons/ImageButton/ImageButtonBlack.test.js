import React from 'react'
import 'jest-styled-components'

import ImageButtonBlack from './ImageButtonBlack'

const { mountWithTheme } = global

describe('(Styled Component) ImageButtonBlack', () => {
  const createImageButtonBlack = (props) => {
    return mountWithTheme(<ImageButtonBlack {...props} />)
  }

  test('matching the snapshot', () => {
    expect(createImageButtonBlack())
    .toMatchSnapshot()
  })
})
