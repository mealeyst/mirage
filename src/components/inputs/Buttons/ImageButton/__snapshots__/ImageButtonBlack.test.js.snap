// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Styled Component) ImageButtonBlack matching the snapshot 1`] = `
.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 35rem;
  overflow: hidden;
}

.c1 > button {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: border-box;
  height: 30px;
  width: 35rem;
  padding: 0 1.5rem;
  text-align: center;
  -webkit-letter-spacing: .1rem;
  -moz-letter-spacing: .1rem;
  -ms-letter-spacing: .1rem;
  letter-spacing: .1rem;
  text-transform: uppercase;
  font-family: "ff-din-web",Helvetica,Arial,sans-serif;
  font-size: 1.4rem;
  font-weight: 500;
  border: none;
}

.c1 > button .arrow {
  margin-left: 1.2rem;
}

.c1 > .borderBottom {
  display: block;
  height: .2rem;
  width: 100%;
  -webkit-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
  -webkit-transition: -webkit-transform 0.25s ease-in-out;
  -webkit-transition: transform 0.25s ease-in-out;
  transition: transform 0.25s ease-in-out;
  position: absolute;
  left: 0;
  bottom: 0;
  z-index: 1;
}

.c1:hover > button {
  cursor: pointer;
}

.c1:hover > .borderBottom {
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  cursor: pointer;
}

.c0 > button {
  background-color: #000000;
  color: #FFFFFF;
}

.c0 > .borderBottom {
  background-color: #F0FF02;
}

<Styled(Styled(CustomButton))>
  <Styled(CustomButton)
    className="c0"
    width="35rem"
  >
    <CustomButton
      className="c0 c1"
      width="35rem"
    >
      <div
        className="c0 c1"
      >
        <button
          width="35rem"
        />
        <span
          className="borderBottom"
        />
      </div>
    </CustomButton>
  </Styled(CustomButton)>
</Styled(Styled(CustomButton))>
`;
